doc/source/how-to-use-clientapp.rst

```diff
@@ -0,0 +1,303 @@
+
+.. code-block:: python
+
+    # instantiate model
+    model = Net()
+
+    # load local training and validation data
+    train_loader, val_loader, _ = load_data()
+
+For simplicity, we have omitted the implementations for :code:`Net()` and :code:`load_data()` , but you can refer to the :code:`quickstart-pytorch` code for similar implementations.
+
+Next, to get the model parameters from the :code:`ServerApp`, we access the :code:`parameters_records` dictionary in the :code:`content` attribute of :code:`Message`, assign value of the :code:`'my_model'` key to a variable, then deserialize it so that we can load the model’s parameter dictionary with the deserialized :code:`state_dict`:
+
+.. code-block:: python
+
+    my_parameters = msg.content.parameters_records['my_model']
+    state_dict = parameters_to_pytorch_state_dict(my_parameters)
+    model.load_state_dict(state_dict=state_dict, strict=True)
+
+Note that in this example, the server sends initial model parameters and training configs for federated learning. It is based on the :doc:`How-to guide for using Flower Driver APIs <how-to-use-driver-api>`. After loading the parameters, we load the configs by accessing the :code:`configs_records` dictionary in the content of the :code:`Message`:
+
+.. code-block:: python
+
+    my_config = msg.content.configs_records['my_config']
+
+For this simple example, we now have enough information to train the model, so let’s do it!
+
+.. code-block:: python
+
+    train_metrics = train_fn(
+        model,
+        train_loader,
+        val_loader,
+        epochs=my_config['epochs'],
+        device='cpu',
+    )
+
+Next, we prepare the updated model parameters and local metrics to be sent to the :code:`ServerApp` for aggregation. To do so, we create a :code:`RecordSet()` and assign to it two things: the serialized parameters to the :code:`parameters_records` attribute dictionary, and the metrics (which are converted to a :code:`MetricsRecord` object) to the :code:`metrics_records` attribute dictionary:
+
+.. code-block:: python
+
+    # Construct reply message carrying updated model parameters and generated metrics
+    reply_content = RecordSet()
+    reply_content.parameters_records['my_model_returned'] = pytorch_to_parameter_record(model)
+    reply_content.metrics_records['train_metrics'] = MetricsRecord(train_metrics)
+
+
+Finally, we return :code:`reply_content` to the :code:`ServerApp` using the :code:`Message.create_reply()` method:
+
+.. code-block:: python
+
+    return msg.create_reply(reply_content)
+
+That’s it! You now have a working Flower :code:`ClientApp` that initializes the model, loads local data, trains the model, and returns the updated model and metrics to the :code:`ServerApp`. Note that the training workflow is dramatically simplified and can be intuitively implemented end-to-end.
+
+For completeness, you can implement the utility functions referenced in the code snippets above as follows:
```

Comment:
Worth stating that soon we'll be providing these utility functions so it becomes trivial to convert a Pythorch/TF/whatever model to a `ParametersRecord` and back.